wp-includes/sqlite-ast/class-wp-sqlite-information-schema-reconstructor.php

```diff
@@ -0,0 +1,427 @@
+				|| str_ends_with( $data_type, 'blob' )
+			) {
+				$cols[] = sprintf(
+					'%s(%d)',
+					$this->quote_sqlite_identifier( $column['name'] ),
+					$data_length ?? $key_length_limit
+				);
+			} else {
+				$cols[] = $this->quote_sqlite_identifier( $column['name'] );
+			}
+		}
+
+		$definition[] = '(' . implode( ', ', $cols ) . ')';
+		return implode( ' ', $definition );
+	}
+
+	/**
+	 * Determine if a column has a default value.
+	 *
+	 * @param  string      $mysql_type    The MySQL data type of the column.
+	 * @param  string|null $default_value The default value of the SQLite column.
+	 * @return bool                       True if the column has a default value, false otherwise.
+	 */
+	private function column_has_default( string $mysql_type, ?string $default_value ): bool {
+		if ( null === $default_value || '' === $default_value ) {
+			return false;
+		}
+		if (
+			"''" === $default_value
+			&& in_array( strtolower( $mysql_type ), array( 'datetime', 'date', 'time', 'timestamp', 'year' ), true )
+		) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Get a MySQL column or index data type from legacy data types cache table.
+	 *
+	 * This method retrieves MySQL column or index data types from a special table
+	 * that was used by an old version of the SQLite driver and that is otherwise
+	 * no longer needed. This is more precise than direct inference from SQLite.
+	 *
+	 * @param  string      $table_name           The table name.
+	 * @param  string      $column_or_index_name The column or index name.
+	 * @return string|null                       The MySQL definition, or null when not found.
+	 */
+	private function get_cached_mysql_data_type( string $table_name, string $column_or_index_name ): ?string {
+		try {
+			$mysql_type = $this->driver->execute_sqlite_query(
+				'SELECT mysql_type FROM _mysql_data_types_cache WHERE `table` = ? AND column_or_index = ?',
+				array( $table_name, $column_or_index_name )
+			)->fetchColumn();
+		} catch ( PDOException $e ) {
+			if ( str_contains( $e->getMessage(), 'no such table' ) ) {
+				return null;
+			}
+			throw $e;
+		}
+		if ( str_ends_with( $mysql_type, ' KEY' ) ) {
+			$mysql_type = substr( $mysql_type, 0, strlen( $mysql_type ) - strlen( ' KEY' ) );
+		}
+		return $mysql_type;
+	}
+
+	/**
+	 * Get a MySQL column type from an SQLite column type.
+	 *
+	 * This method converts an SQLite column type to a MySQL column type as per
+	 * the SQLite column type affinity rules:
+	 *   https://sqlite.org/datatype3.html#determination_of_column_affinity
+	 *
+	 * @param  string $column_type The SQLite column type.
+	 * @return string              The MySQL column type.
+	 */
+	private function get_mysql_data_type( string $column_type ): string {
+		$type = strtoupper( $column_type );
+		if ( str_contains( $type, 'INT' ) ) {
+			return 'int';
+		}
+		if ( str_contains( $type, 'TEXT' ) || str_contains( $type, 'CHAR' ) || str_contains( $type, 'CLOB' ) ) {
```

Comment:
This came up for me as I read it: CLOB? Or BLOB? Also, in another line of this file we do strtolower and test for str_ends_with, let's be consistent here if that makes sense.

Then I've remembered sqlite uses substring to match a data type. And clob is actually a char type. Still, let's document that inline. These are all non-obvious behaviors